clients/src/main/java/org/apache/kafka/common/requests/FetchRequest.java

```diff
@@ -144,18 +146,28 @@ public static class Builder extends AbstractRequest.Builder<FetchRequest> {
 
         public static Builder forConsumer(short maxVersion, int maxWait, int minBytes, Map<TopicPartition, PartitionData> fetchData) {
             return new Builder(ApiKeys.FETCH.oldestVersion(), maxVersion,
-                CONSUMER_REPLICA_ID, maxWait, minBytes, fetchData);
+                CONSUMER_REPLICA_ID,  -1, maxWait, minBytes, fetchData);
         }
 
-        public static Builder forReplica(short allowedVersion, int replicaId, int maxWait, int minBytes,
+        public static Builder forReplica(short allowedVersion, int replicaId, long replicaEpoch, int maxWait, int minBytes,
                                          Map<TopicPartition, PartitionData> fetchData) {
-            return new Builder(allowedVersion, allowedVersion, replicaId, maxWait, minBytes, fetchData);
+            return new Builder(allowedVersion, allowedVersion, replicaId, replicaEpoch, maxWait, minBytes, fetchData);
         }
 
-        public Builder(short minVersion, short maxVersion, int replicaId, int maxWait, int minBytes,
+        public Builder(short minVersion, short maxVersion, int replicaId, long replicaEpoch, int maxWait, int minBytes,
                        Map<TopicPartition, PartitionData> fetchData) {
             super(ApiKeys.FETCH, minVersion, maxVersion);
             this.replicaId = replicaId;
+            this.replicaEpoch = replicaEpoch;
+            this.maxWait = maxWait;
+            this.minBytes = minBytes;
+            this.toFetch = fetchData;
+        }
+        public Builder(short minVersion, short maxVersion, int replicaId, int maxWait, int minBytes,
```

Comment:
I must be missing something, but isn't this constructor and the one above line 157 the same (that is only differing in the order of definition of their arguments)? Is this overload intended?

Reply:
Good catch. the second one should be removed.